src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php

```diff
@@ -246,6 +246,14 @@ public function prepare_item_for_response( $item, $request ) {
 			$data['rest_namespace'] = $namespace;
 		}
 
+		if ( rest_is_field_included( 'template', $fields ) && ! empty( $post_type->template ) ) {
+			$data['template'] = $post_type->template;
+		}
+
+		if ( rest_is_field_included( 'template_lock', $fields ) && ! empty( $post_type->template_lock ) && false !== $post_type->template_lock ) {
```

Comment:
```suggestion
		if ( rest_is_field_included( 'template_lock', $fields ) && ! empty( $post_type->template_lock ) ) {
```

`empty` already checks for the value being false.